auslib/blobs/base.py

```diff
@@ -78,9 +78,17 @@ def merge_dicts(ancestor, left, right):
      * A type mismatch of unicode vs string is OK as long as the text is the same
        (Any other type mismatches result in a failure to merge.)
     """
+    # We can't use "logging" directly, because it ignores our custom code in log.py
+    log = logging.getLogger(__name__)
     result = {}
     dicts = (ancestor, left, right)
     for key in set(key for d in dicts for key in d.keys()):
+        # Extra logging information to help debug https://bugzilla.mozilla.org/show_bug.cgi?id=1501167
+        # This is a very large message, so we limit it as much as possible to reduce spam.
+        if key == "completes" and ancestor.get("appVersion") and "a1" not in ancestor["appVersion"]:
+            log.warning("Ancestor is: %s", ancestor.get(key))
+            log.warning("Left is: %s", left.get(key))
+            log.warning("Right is: %s", right.get(key))
```

Comment:
For brevity, perhaps the left and right could be expressed as diffs against the ancestor.

Reply:
Hmmm, I think we can get most of what we need from earlier messages (which will contain the same `requestid`. Specifically, the messages around https://github.com/mozilla/balrog/blob/03976a24c3bc3d7ad696e846311356f80e1bdde6/auslib/db.py#L2015 give us the ancestor data version. I think it would be good to print the last two messages (ancestor change id, tip data version) for the success case for now, though. That should give us all the information we need to decode which is ancestor/left/right.